Regular Expressions/RegExp Common.sublime-syntax

```diff
@@ -0,0 +1,275 @@
+%YAML 1.2
+---
+name: Regular Expression
+hidden: true
+scope: source.regexp-common
```

Comment:
Can we avoid using dashes in scope names? It's barely used and doesn't feel that good.

We used `common` in Git Formats to specify a file which is re-used by many but does not work as standalone syntax.

If I understand your PR comment, this one is a standalone PCRE syntax without boost specific stuff?

If so, I'd suggest `source.regexp.pcre` for this file.

Once we've learned about commonly used stuff between several dialects (python, php, c#, ...) we might introduce a `source.regexp.basic` or even a `source.regexp.common` for shared contexts?

Reply:
> Once we've learned about commonly used stuff between several dialects (python, php, c#, ...) we might introduce a `source.regexp.basic` or even a `source.regexp.common` for shared contexts?

This is actually already what I was trying to do, sorry if something I wrote was misleading - `Regexp Common.sublime-syntax` is supposed to represent what is common in all regex engines, so I think `source.regexp.basic` would perhaps make sense.

My idea was to maybe also have *unused* contexts for some things which are not common for all dialects, but shared in some. These could then be included/referenced by the extended syntax for the dialects which use them.